### DISCUSSION TITLE
No update on PG_BAGE when popup is active

### DISCUSSION THREAD

**Discussion by umsmin** [ImpulseAdventure/GUIslice#496] [Q&A]
I have a clock as an updating text on PG_BASE so that's shown every time. But when one of my Popups are running the the time is frozen. No update at all.

Thanks 
Uli

**Comment by umsmin** [comment]
@ImpulseAdventure 
because no one have maybe this problem. I talked about this problem with Pconti31 he also see this. 
> I have a clock as an updating text on PG_BASE so that's shown every time. But when one of my Popups are running the the time is frozen. No update at all.
> 
> Thanks Uli

**Comment by Pconti31** [comment]
I think I found the solution to your problem. In looking over GUIslice.c 
I found this comment inside void gslc_PopupShow()

```
  // If modal dialog selected, then deactivate other pages in stack
  // If modeless dialog selected, then don't deactivate other pages in stack

```

Now the Builder outputs popup pages as modal dialogs so your code looks something like this:

```
      case E_ELEM_BTN4:
        gslc_PopupShow(&m_gui, E_PG_POPUP1, true);
        break;
```
changing it to:
```
      case E_ELEM_BTN4:
        gslc_PopupShow(&m_gui, E_PG_POPUP1, false);
        break;
```

Makes the popup dialog modeless and the time will update!

Paul--

**Comment by umsmin** [comment]
@Pconti31 You are the BEST. It works perfectly. So I have to change it manually in the xxx.ino file. The good is the Guislice-Builder don't change it when I edit something. 
I think to make a switchbox in the part of POPUP in Guislice-Builder will be a good feature.

Thanks

Uli

**Comment by Pconti31** [comment]
@umsmin I agree that adding an enhancement to allow the builder to have a modeless property on popup pages would be nice but due the current design it would be hard to do.   So unless someone offers to help with coding its not going to happen.
Sorry.
Paul--
